index.d.ts

```diff
@@ -0,0 +1,114 @@
+    [other: string]: any;
+}
+
+export type DynamicMetaFunction = (req: Request, res: Response, err: Error) => object;
+export type DynamicLevelFunction = (req: Request, res: Response, err: Error) => string;
+export type RequestFilter = (req: FilterRequest, propName: string) => any;
+export type ResponseFilter = (res: FilterResponse, propName: string) => any;
+export type RouteFilter = (req: Request, res: Response) => boolean;
+export type MessageTemplate = string | ((req: Request, res: Response) => string);
+
+export interface BaseLoggerOptions {
+    baseMeta?: object;
+    bodyBlacklist?: string[];
+    bodyWhitelist?: string[];
+    colorize?: boolean;
+    dynamicMeta?: DynamicMetaFunction;
+    expressFormat?: boolean;
+    format?: Format;
+    ignoreRoute?: RouteFilter;
+    ignoredRoutes?: string[];
+    level?: string | DynamicLevelFunction;
+    meta?: boolean;
+    metaField?: string;
+    requestField?: string;
+    responseField?: string;
+    msg?: MessageTemplate;
+    requestFilter?: RequestFilter;
+    requestWhitelist?: string[];
+    responseFilter?: ResponseFilter;
+    responseWhitelist?: string[];
+    skip?: RouteFilter;
+    statusLevels?: {
+        error?: string;
+        success?: string;
+        warn?: string;
+    };
+}
+
+export interface LoggerOptionsWithTransports extends BaseLoggerOptions {
+    transports: Transport[];
+}
+
+export interface LoggerOptionsWithWinstonInstance extends BaseLoggerOptions {
+    winstonInstance: winston.Logger;
+}
+
+export type LoggerOptions = LoggerOptionsWithTransports | LoggerOptionsWithWinstonInstance;
+
+export function logger(options: LoggerOptions): Handler;
+
+export interface BaseErrorLoggerOptions {
+    baseMeta?: object;
+    dynamicMeta?: DynamicMetaFunction;
+    format?: Format;
+    level?: string | DynamicLevelFunction;
+    meta?: boolean;
+    metaField?: string;
+    requestField?: string;
+    msg?: MessageTemplate;
+    requestFilter?: RequestFilter;
+    requestWhitelist?: string[];
+}
+
+export interface ErrorLoggerOptionsWithTransports extends BaseErrorLoggerOptions {
+    transports: Transport[];
+}
+
+export interface ErrorLoggerOptionsWithWinstonInstance extends BaseErrorLoggerOptions {
+    winstonInstance: winston.Logger;
+}
+
+export type ErrorLoggerOptions = ErrorLoggerOptionsWithTransports | ErrorLoggerOptionsWithWinstonInstance;
+
+export function errorLogger(options: ErrorLoggerOptions): ErrorRequestHandler;
+
+export let requestWhitelist: string[];
+
+export let bodyWhitelist: string[];
+
+export let bodyBlacklist: string[];
+
+export let responseWhitelist: string[];
+
+export let ignoredRoutes: string[];
+
+export let defaultRequestFilter: RequestFilter;
+
+export let defaultResponseFilter: ResponseFilter;
+
+export function defaultSkip(): boolean;
+
+export interface ExpressWinstonRequest extends Request {
+    _routeWhitelists: {
+        body: string[];
+        req: string[];
+        res: string[];
+    };
+}
```

Comment:
I don't want to add another stopper for this PR but I believe the type definitions are in definitely typed right now.

summoning @golopot

Reply:
Yes that's absolutely right. However using DefinitelyTyped to publish your types increases overhead  of using the library and adds complexity of compatibility between implementation and type definitions.  DT even has a part of their readme dedicated to removing a package from DT when its maintainers add the types to the package.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#removing-a-package

If the package maintainer is willing to include the types it helps accomplish two things: 
1. You do not need to add two dependencies to get autosense and TypeScript compilation to function on a project.  This means that transitive dependencies also get the types.
2. The implementation of the type definition gets versioned with the definition itself so when you add new functionality, a second PR to DefinitelyTyped is not required and there is not version conflicts between updating the types and the implementation.

I can't see any real reason to use DefinitelyTyped other than the small decrease in the size of your published NPM package however since there was no "files" declared in the original package.json nor a .npmignore, multiple unnecessary files were being packaged with the deployed library (also fixed in this PR).  Additionally since "express-winston" is a "backend" library (i.e. it would never run in a browser since Express doesn't run in a browser), the reducing size of the library (especially less than 1k difference) is not helpful.